### ISSUE TITLE
Add reward relabelling to `train_preference_comparisons` and `train_adversarial` for off-policy algorithms

### ISSUE THREAD

**Issue by yawen-d** [HumanCompatibleAI/imitation#458]
## Suggested Feature
- Allow `AgentTrainer` to relabel the learned rewards of all transitions in the replay buffer after the reward model is updated for off-policy algorithms.
- This feature applies to both `train_adversarial.py` and `train_preference_comparisons.py` when an off-policy algorithm is used as their RL component.

## Motivation
- Reward relabelling was used in the paper
[PEBBLE: Feedback-Efficient Interactive Reinforcement Learning via Relabeling Experience and Unsupervised Pre-training](https://arxiv.org/pdf/2106.05091.pdf) to mitigate the effects of a non-stationary reward function.

**Comment by yawen-d** [comment]
- Preference Comparisons + SAC (w/ Reward Relabeling) outperforms (w/o Reward Relabeling)
![Screenshot 2022-07-06 at 11 14 20](https://user-images.githubusercontent.com/35738259/177616118-9b95e9ad-0995-441c-86ae-4768dc1510bc.png)

- Preference Comparisons + SAC (w/ Reward Relabeling) significantly outperforms (w/o Reward Relabeling)
![Screenshot 2022-07-06 at 11 16 13](https://user-images.githubusercontent.com/35738259/177616370-35a61562-7409-4ed3-9187-4602c2f24055.png)

**Comment by AdamGleave** [comment]
Experimental results definitely show the benefit of this feature, plus the results in prior work (PEBBLE), so I'm in favour of us having an implementation :)

**Comment by yawen-d** [comment]
Close issue because #459 has been merged.
